jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostEstimator.scala

```diff
@@ -107,18 +107,33 @@ class XGBoostEstimator private[spark](
     }
   }
 
+  private def ensureColumns(trainingSet: Dataset[_]): Dataset[_] = {
```

Comment:
I mean, when they do not provide baseMargin, we just move forward with the idea that baseMargin is not provided